client/src/components/UserPassword.js

```diff
@@ -1,39 +1,48 @@
 // @flow
 
 import type { EditContainerProps } from '@performant-software/shared-components/types';
-import React from 'react';
+import React, { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 import { Form, Message } from 'semantic-ui-react';
+import UserUtils from '../utils/User';
 
-type Props = EditContainerProps;
+type Props = EditContainerProps & {
+  email: string
+};
 
 const UserPassword = (props: Props) => {
   const { t } = useTranslation();
 
-  return (
-    <>
-      <Message
-        content={t('UserPassword.messages.password.content')}
-        header={t('UserPassword.messages.password.header')}
-      />
-      <Form.Input
-        error={props.isError('password')}
-        label={t('UserPassword.labels.password')}
-        onChange={props.onTextInputChange.bind(this, 'password')}
-        required={props.isRequired('password')}
-        type='password'
-        value={props.item.password || ''}
-      />
-      <Form.Input
-        error={props.isError('password_confirmation')}
-        label={t('UserPassword.labels.passwordConfirmation')}
-        onChange={props.onTextInputChange.bind(this, 'password_confirmation')}
-        required={props.isRequired('password_confirmation')}
-        type='password'
-        value={props.item.password_confirmation || ''}
-      />
-    </>
-  );
+  const showPasswordFields = useMemo(() => UserUtils.showPasswordFields(props.email), [props.email]);
```

Comment:
I think we should pull this logic out of the component as well. It would make the `UserPassword` component reusable by not embedding any of the SSO logic. I don't think we necessarily need the "can't change password" message.

```
{ UserUtils.showPasswordFields(email) && (
  <UserPassword 
    ...
  />
)}
```